### PR TITLE
Release/2.1.0

### PR DESCRIPTION
**Pull request by AnonymousWP**
- Introduced `OllamaManager` API class with `get`, `delete`, and `put` methods for managing Ollama models.
- Enhanced `WafDetector` and `CMSDetector` APIs with improved error handling and logging.
- Added organization handling in `AddTarget` and bulk target addition.
- Updated **Docker** and **Makefile** configurations for better build and deployment processes.
- Implemented new logging configurations and improved settings management.
- Added support for **Lark** notifications and improved API key management in the UI.
- Refactored various paths and configurations to use environment variables and Path objects for better flexibility.
- Enhanced scan initiation and scheduling with user context and improved error handling.
- Added new utility functions and refactored existing ones for better code organization and readability.

### CODE REVIEW

**Comment by yarysp** [comment]
Latest modification from master has been merged.
No more conflict (solved by hand :cry: )
Need tests

**Comment by psyray** [comment]
Rebased from upstream

**Comment by AnonymousWP** [comment]
Please link all the issues too, so that when we merge the 2.1.0 branch, corresponding issues get automatically closed.

image

File: web/reNgine/common_func.py [hunk 1]
```diff
@@ -1055,3 +1160,193 @@
...
+    Args:
+        command (str or list): The command to execute.
+        shell (bool): Whether to use shell execution.
+        cwd (str): The working directory for the command.
+
+    Returns:
+        subprocess.Popen: The Popen object for the executed command.
+    """
+    return subprocess.Popen(
+        command,
```
[the remark this answers is not in the document]
Comment (psyray) [reply web/reNgine/common_func.py:1275]: **shell** is always set to **False** when the `execute_command` is called 
https://github.com/Security-Tools-Alliance/rengine-ng/blob/9c40886b7059b8b2981f47916975847b2e8b5634/web/reNgine/tasks.py#L4187

And **shlex.split** is used in the `prepare_command` function
https://github.com/Security-Tools-Alliance/rengine-ng/blob/9c40886b7059b8b2981f47916975847b2e8b5634/web/reNgine/common_func.py#L1194-L1205

`prepare_command` function is systematically called before running a command so the command string is always safely parsed into a list of arguments
https://github.com/Security-Tools-Alliance/rengine-ng/blob/9c40886b7059b8b2981f47916975847b2e8b5634/web/reNgine/tasks.py#L4203-L4208

File: web/api/serializers.py [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-														   naturaltime)
+from django.contrib.humanize.templatetags.humanize import (naturalday, naturaltime)
 from django.db.models import F, JSONField, Value
 from recon_note.models import *
 from reNgine.common_func import *
 from rest_framework import serializers
 from scanEngine.models import *
 from startScan.models import *
 from targetApp.models import *
+from dashboard.models import *
```
[the remark this answers is not in the document]
Comment (psyray) [reply web/api/serializers.py:11]: import * will be fixed later when refactoring things

File: web/dashboard/tests/test_dashboard.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
+"""
+This file contains the test cases for the dashboard views.
+"""
+import json
+from unittest.mock import patch, MagicMock
+from django.urls import reverse
+from utils.test_base import BaseTestCase
+from dashboard.views import admin_interface_update
+from dashboard.models import Project
```
[the remark this answers is not in the document]
Comment (psyray) [reply web/dashboard/tests/test_dashboard.py:9]: not used import will be fixed later when refactoring things

**Review by psyray — APPROVED** [review]

**Pull request merged**